src/components/layout/sync-button.tsx

```diff
@@ -0,0 +1,158 @@
+import { Button, Box, LinearProgress, Snackbar, Alert } from "@mui/material";
+import { DateTime } from "luxon";
+import { SyntheticEvent, useState } from "react";
+import { useApi } from "../../hooks/use-api";
+import { errorAtom } from "../../atoms/error";
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+import SyncDialog from "../contexts/sync-handler";
+import strings from "../../localization/strings";
+import { dailyEntriesAtom, personTotalTimeAtom, personsAtom } from "../../atoms/person";
+import { DailyEntry, Person } from "../../generated/client";
+import config from "../../app/config";
+import { userProfileAtom } from "../../atoms/auth";
+
+/**
+ * Sync button component
+ */
+const SyncButton = () => {
+  const { synchronizeApi } = useApi();
+  const setError = useSetAtom(errorAtom);
+  const yesterday: DateTime = DateTime.now().minus({ days: 1 });
+  const [syncStartDate, setSyncStartDate] = useState(yesterday);
+  const [syncing, setSyncing] = useState(false);
+  const [syncSuccess, setSyncSuccess] = useState(false);
+  const [syncHandlerOpen, setSyncHandlerOpen] = useState(false);
+  const [dailyEntries, setDailyEntries] = useAtom(dailyEntriesAtom);
+  const [persons, setPersons] = useAtom(personsAtom);
+  const { personsApi, dailyEntriesApi } = useApi();
+  const userProfile = useAtomValue(userProfileAtom);
+  const personTotalTime = useAtomValue(personTotalTimeAtom);
+
+  /**
+   * fetching persons and daily entries
+   */
+  const fetchPersonsAndEntries = async () => {
```

Comment:
This method does two things. In reactive context, you should instead have a separate method for dailyEntries in useEffect what depends on persons